### PR TITLE
ROC-3315: Handle 409 status with request retries

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-3315

### Change description

An update to `ClaimStoreClient` that will make it return a saved claim by externalId if the initial save times out and a following retry results in 409 Conflict response.

### Work checklist

- [X] Unit tests added where applicable

### Developer self-QA run statement

- [X] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/main/app/claims/claimStoreClient.ts [hunk 1]
```diff
@@ -15,21 +16,35 @@ export const claimStoreApiUrl: string = `${claimApiBaseUrl}/claims`
...
-          Authorization: `Bearer ${user.bearerToken}`
+          Authorization: `Bearer ${claimant.bearerToken}`
+        }
+      })
+      .then(claim => {
+        return new Claim().deserialize(claim)
+      })
+      .catch((err) => {
+        if (err.statusCode === HttpStatus.CONFLICT) {
+          return this.retrieveByExternalId(draft.document.externalId, claimant)
```
Comment (geek-soft) [inline src/main/app/claims/claimStoreClient.ts]: I would love to have some logs here stating that conflict occurred and we retrieved claim from data store.

File: src/main/app/claims/claimStoreClient.ts [hunk 2]
```diff
@@ -15,21 +16,35 @@ export const claimStoreApiUrl: string = `${claimApiBaseUrl}/claims`
...
+        } else {
+          throw err
```
Comment (pragnesh-patel) [inline src/main/app/claims/claimStoreClient.ts]: As @geek-soft suggested logging here would be beneficial for support

Comment (timja-kainos) [reply src/main/app/claims/claimStoreClient.ts]: logging will already happen here @pragnesh-patel , damian suggested 2 lines up

Comment (pragnesh-patel) [reply src/main/app/claims/claimStoreClient.ts]: ok...

**Pull request merged**